src/lxc/tools/lxc_execute.c

```diff
@@ -73,6 +77,10 @@ static const struct option my_longopts[] = {
 	{"define", required_argument, 0, 's'},
 	{"uid", required_argument, 0, 'u'},
 	{"gid", required_argument, 0, 'g'},
+	{"share-net", required_argument, 0, OPT_SHARE_NET},
+	{"share-ipc", required_argument, 0, OPT_SHARE_IPC},
+	{"share-uts", required_argument, 0, OPT_SHARE_UTS},
+	{"share-pid", required_argument, 0, OPT_SHARE_PID},
```

Comment:
Ok, so here I get scared, though, because if you look at src/lxc/tools/lxc_init.c, when it gets a signal it likes to kill -1.  So if you share-pid with host pid 1, then sending sigterm to the container will try to send that to the host init.

Now for unpriv containers that shoudl be fine as it won't be allowed to kill -1, but for privileged containers that's a bad thing.

Not sure what to do about it though.  Just let the user shoot themselves in the foot?

Reply:
Maybe we update lxc_init to not kill -1 if it's share-pid?

Reply:
@hallyn, if there's a sensible way to do this then sure.

Reply:
It would have to detect that its vpid != 1.

Reply:
Ok, I'll add a patch on top to do this.